encoding/gzip/gzip.go

```diff
@@ -107,6 +109,20 @@ func (z *reader) Read(p []byte) (n int, err error) {
 	return n, err
 }
 
+// RFC1952 specifies that the last four bytes "contains the size of
+// the original (uncompressed) input data modulo 2^32."
+func (c *compressor) DecompressedSize(buf []byte, maxSize int) (int, error) {
+	if int64(maxSize) > int64(math.MaxUint32) {
```

Comment:
I was thinking other compressors wouldn’t have the same limitation. The 4-byte size is specific to gzip so gzip has to check it.

Reply:
This particular check is actually not necessary.  maxSize, set by gRPC, will never be > 2GB.  But even ignoring that, the `Compressor` shouldn't need to worry about what gRPC will consider acceptable; it should just provide a number and gRPC should determine if it is OK.

(Your comment seems to be about the check below, however?)

Reply:
To clarify, the case I was thinking about when I wrote that code was that the real uncompressed size is, say, 2^32 + 1000, so it will be reported as 1000.
So, I thought, if you are expecting >4GB messages then we shouldn't try to report the size.
But of course 1000 is useless in every case.

Now I'm thinking something like a `LimitWriter` would fit the bill better.